File: appli-android/app/src/main/java/com/example/sportapp/core/data/model/MuscleGoal.kt
package com.example.sportapp.core.data.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.example.sportapp.core.utils.CustomDateUtils.getNowISO8601
import java.util.UUID

@Entity(
    tableName = "muscle_goals",
    indices = [androidx.room.Index(value = ["uuid"], unique = true)],
    foreignKeys = [
        androidx.room.ForeignKey(
            entity = Muscle::class,
            parentColumns = ["uuid"],
            childColumns = ["muscle_uuid"],
            onDelete = androidx.room.ForeignKey.CASCADE
        )
    ],
)
data class MuscleGoal(
    @PrimaryKey
    @ColumnInfo(name = "uuid") val uuid: String,

    @ColumnInfo(name = "user_id") val userId: Int,
    @ColumnInfo(name = "muscle_uuid") val muscleUUID: String,
    val priority: String,
    val done: Int,
    val target: String,
    @ColumnInfo(name = "week_iso") val weekISO: String,
    val status: String = "IN_PROGRESS", // Can be "DONE", "IN_PROGRESS", "NOT_STARTED", "SKIPPED"
    @ColumnInfo(name = "added_manually") val addedManually: Boolean = false,

    @ColumnInfo(defaultValue = "0") val synced: Boolean = false,
    @ColumnInfo(defaultValue = "0") val pendingDeletion: Boolean = false,

    @ColumnInfo(name = "updated_at") val updatedAt: String? = getNowISO8601(),
)

File: serveur/app/models/muscle_goal.py
from sqlalchemy import Column, Integer, String, Boolean, ForeignKey, DateTime, func
from app.database import Base
import uuid

class MuscleGoal(Base):
    __tablename__ = "muscle_goals"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    muscle_uuid = Column(String, ForeignKey("muscles.uuid", ondelete="CASCADE"), nullable=False, index=True)

    uuid = Column(String, nullable=False, default=lambda: str(uuid.uuid4()), unique=True)

    priority = Column(String, nullable=False)
    done = Column(Integer, default=0, nullable=False)
    target = Column(String, nullable=False)
    week_iso = Column(String, nullable=False)
    status = Column(String, default="IN_PROGRESS", nullable=False)
    added_manually = Column(Boolean, default=False, nullable=False)

    updated_at = Column(DateTime(timezone=True), nullable=True)
